app/src/lib/stores/git-store.ts

```diff
@@ -1459,13 +1460,21 @@ export class GitStore extends BaseStore {
     await queueWorkHigh(files, async file => {
       const foundSubmodule = submodules.some(s => s.path === file.path)
 
-      if (file.status.kind !== AppFileStatusKind.Deleted && !foundSubmodule) {
+      if (
+        file.status.kind !== AppFileStatusKind.Deleted &&
+        !foundSubmodule &&
+        moveToTrash
+      ) {
         // N.B. moveItemToTrash can take a fair bit of time which is why we're
         // running it inside this work queue that spreads out the calls across
         // as many animation frames as it needs to.
-        await this.shell.moveItemToTrash(
-          Path.resolve(this.repository.path, file.path)
-        )
+        try {
+          await this.shell.moveItemToTrash(
+            Path.resolve(this.repository.path, file.path)
+          )
+        } catch (e) {
+          throw new ErrorWithMetadata(e, {})
```

Comment:
Using  `new ErrorWithMetadata(e, {})` as hacky way to distinguish an error thrown by the `moveItemToTrash` method somewhere else in this function...  needs more thought into this approach.... Either moving the `moveItemToTrash` up to the app-store (but then it's not in that queue) and handling the error there.. or some sort of direct return type indicating this specific failure.

Reply:
(Ignore from first stab)